File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDetailProductTaxMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDetailProductTaxDTO;
import com.robinfood.core.entities.OrderDetailProductTaxEntity;

public final class OrderDetailProductTaxMappers {

    private OrderDetailProductTaxMappers() {
        // this constructor is empty because it is a mapper class
    }

    public static OrderDetailProductTaxDTO toOrderDetailProductTaxMappers(
            OrderDetailProductTaxEntity orderDetailProductTaxEntity
    ) {
        return new OrderDetailProductTaxDTO(
                orderDetailProductTaxEntity.getFamilyTypeId(),
                orderDetailProductTaxEntity.getId(),
                orderDetailProductTaxEntity.getPrice(),
                orderDetailProductTaxEntity.getTaxTypeId(),
                orderDetailProductTaxEntity.getTaxTypeName(),
                orderDetailProductTaxEntity.getValue()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDetailProductTaxMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.constants.GlobalConstants.DEFAULT_INTEGER_VALUE
import com.robinfood.core.dtos.OrderDetailProductTaxDTO
import com.robinfood.core.entities.OrderDetailProductTaxEntity
import java.math.BigDecimal

fun OrderDetailProductTaxEntity.toOrderDetailProductTaxDTO(): OrderDetailProductTaxDTO? {
    return if (id == null) {
        null
    } else {
        return OrderDetailProductTaxDTO(
                familyTypeId = familyTypeId ?: DEFAULT_INTEGER_VALUE,
                id = id,
                price = price ?: BigDecimal.ZERO,
                value = value ?: BigDecimal.ZERO
        )
    }
}